server/src/main/java/org/spine3/server/command/CommandBus.java

```diff
@@ -246,32 +270,31 @@ private void dispatch(Command command) {
 
     private void invokeHandler(Message msg, CommandContext context) {
         final CommandClass commandClass = CommandClass.of(msg);
-        final CommandHandler handler = getHandler(commandClass);
+        final CommandHandler handler = handlerRegistry.getHandler(commandClass);
         final CommandId commandId = context.getCommandId();
         try {
             handler.handle(msg, context);
             commandStatusService.setOk(commandId);
         } catch (InvocationTargetException e) {
-            final Throwable cause = e.getCause();
-            //noinspection ChainOfInstanceofChecks
-            if (cause instanceof Exception) {
-                final Exception exception = (Exception) cause;
-                problemLog.errorHandling(exception, msg, commandId);
-                commandStatusService.setToError(commandId, exception);
-            } else if (cause instanceof FailureThrowable){
-                final FailureThrowable failure = (FailureThrowable) cause;
-                problemLog.failureHandling(failure, msg, commandId);
-                commandStatusService.setToFailure(commandId, failure);
-            } else {
-                problemLog.errorHandlingUnknown(cause, msg, commandId);
-                commandStatusService.setToError(commandId, Errors.fromThrowable(cause));
-            }
+            onHandlerException(msg, commandId, e);
         }
     }
 
-    private CommandHandler getHandler(CommandClass commandClass) {
-        final CommandHandler handler = handlerRegistry.getHandler(commandClass);
-        return handler;
+    private void onHandlerException(Message msg, CommandId commandId, InvocationTargetException e) {
```

Comment:
It would be better to pass `Throwable` to this method. It would make the purpose of the method more clear.